scvi/train/_trainingplans.py

```diff
@@ -516,13 +520,55 @@ def validation_step(self, batch, batch_idx, optimizer_idx=0):
         _, _, scvi_losses = self.forward(full_dataset, loss_kwargs=input_kwargs)
         loss = scvi_losses.loss
         reconstruction_loss = scvi_losses.reconstruction_loss
-        return {
+
+        loss_dict = {
             "loss": loss,
             "reconstruction_loss_sum": reconstruction_loss.sum(),
             "kl_local_sum": scvi_losses.kl_local.sum(),
             "kl_global": scvi_losses.kl_global,
             "n_obs": reconstruction_loss.shape[0],
         }
+        if hasattr(scvi_losses, "classification_loss"):
+            loss_dict["classification_loss"] = scvi_losses.classification_loss
+            loss_dict["n_labelled_tensors"] = scvi_losses.n_labelled_tensors
+        return loss_dict
+
+    def training_epoch_end(self, outputs):
+        super().training_epoch_end(outputs)
+        log_classification_loss = False
```

Comment:
what is this line doing?

Reply:
log as in calling self.log. If there is classification loss to log, this flag is set to true and the loss is logged.